ksqldb-engine/src/main/java/io/confluent/ksql/internal/StorageUtilizationMetricsReporter.java

```diff
@@ -65,6 +68,7 @@ public synchronized void configure(final Map<String, ?> map) {
         map.get(KsqlConfig.KSQL_INTERNAL_METRICS_CONFIG)
     );
     this.metricsSeen = new HashMap<>();
+    this.numberStatefulTasks = new AtomicInteger(0);
```

Comment:
This is a static, so we shouldn't re-initialize for each instance. We should just initialize this statically (e.g. `private static AtomicInteger numberStatefulTasks = new AtomicInteger(0);`) or from `configureShared`

Reply:
just changed it to be initialized statically, updated the gauge as well